packages/lesswrong/components/hooks/useRejectContent.tsx

```diff
@@ -0,0 +1,37 @@
+import { useCallback } from 'react';
+import { useCurrentUser } from '../common/withUser';
+import { useUpdate } from '../../lib/crud/withUpdate';
+
+export type RejectContentParams = {
+  collectionName: "Posts",
+  content: SunshinePostsList
+} | {
+  collectionName: "Comments",
+  content: CommentsList | CommentsListWithParentMetadata
+}
+
+export function useRejectContent ({collectionName, content}: RejectContentParams) {
+  const fragmentName = collectionName === "Posts" ? "SunshinePostsList" : "CommentsListWithParentMetadata"
+  const { mutate: updateContent } = useUpdate({
+    collectionName,
+    fragmentName
+  });
+  
+  const rejectContent = useCallback((reason: string) => {
```

Comment:
I think it would be good to have an object that stores the reason ids so we can do analytics on which ones we most often use.